### PR TITLE
feat: configuring jest and create some tests

### PR DESCRIPTION
**Pull request by ftonato**
- removing unnecessary components/code
- adding dependencies to test
- configuring the dependencies (eslint, jest, tsconfig, babel)
- adding three initial tests using jest / @testing-library/react

References: #4

### CODE REVIEW

File: test/pages/index.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  })
+
+  it('should display a correct counter when the value changes', () => {
+    const user = `\ncc @sseraphini`
+    const encode = (str) => encodeURIComponent(str)
+
+    render(<Home />)
+    const textarea = screen.getByRole('textbox')
+
+    const badge = screen.getByTestId('counter')
```
Comment (sibelius) [inline test/pages/index.test.js:49]: can we avoid testId ?

getByText would be enough here

File: src/pages/index.tsx [hunk 2]
```diff
@@ -51,7 +51,7 @@ const Home: NextPage = () => {
             onChange={(e) => setText(e.target.value)}
           />
 
-          <Badge maxW="fit-content" colorScheme={counter < 0 ? 'red' : ''}>
+          <Badge data-testid="counter" maxW="fit-content" colorScheme={counter < 0 ? 'red' : ''}>
```
Comment (sibelius) [inline src/pages/index.tsx:54]: not needed

File: test/pages/index.test.js (hunk 1 above)
Comment (ftonato) [reply test/pages/index.test.js:49]: I can change it, why do you prefer to avoid the use of `testId's`?

Comment (sibelius) [reply test/pages/index.test.js:49]: https://betterprogramming.pub/why-you-should-avoid-testing-react-components-with-test-ids-ee50d20d37d2

you should test like the end user, end user can't "see" test-id

Comment (ftonato) [reply test/pages/index.test.js]: they can't see also the tests ^^

**Pull request merged**